### DISCUSSION TITLE
How do I...

### DISCUSSION THREAD

**Discussion by dtpoirot** [ImageMagick/ImageMagick#7131] [Digital Image Processing] [answered]
Hi, I'm new to ImageMagick. All my image processing was done in university in machine vision applications...

I have a color PNG file of a scanned black and white photo from 1935. How do I punch it up for screen printing (for a t-shirt)? 

I want to convert the RGB file to greyscale. I want to normalize the histogram. I want to clip and stretch the histogram. I may want to flatten the histogram. I then want to convert it to a pixelated halftone.

Surprisingly, I know what all those words mean! ;-) I want to bring up the details, get rid of the black-black and the white-white, optimize contrast, and then convert to what an old-time newspaper image looks like. Right?

How much of this can I do in IM? Please help me Mr/Ms Wizard!!!

**Comment by dtpoirot** [comment]
...by the way, Adaptive Histogram Equalization (AHE) looks very pretty!

The screen print on a t-shirt presents interesting obstacles. I am using a photo-resist on the fine line (120 threads per inch) nylon screen mesh, so get excellent detail. 

The challenge comes when printing the image for photo exposure. I am using one color. Using a direct print from the limited shade, low contrast photo from 1935, almost all of the contrast goes away.

**Reply by fmw42** [reply]
How about posting an example image

**Reply by dtpoirot** [reply]
[First class of Lee College graduates, graduated May 24, 1935](https://texashistory.unt.edu/ark:/67531/metapth1204618/) hosted by [The Portal to Texas History](https://texashistory.unt.edu/)

This photograph has been made publicly available for use in research, teaching, and private study by the [Lee College ](https://texashistory.unt.edu/explore/partners/LCBT/)in partnership with [The Portal to Texas History](https://texashistory.unt.edu/), a digital repository hosted by the [University of North Texas Libraries](https://library.unt.edu/).

**Reply by dtpoirot** [reply]
Holy Smokes! FMW42 has a lifetime of scripts that do amazing things!!!

I had a very full day, yesterday, installing ImageMagick on my 2012 MacBook Pro. For some reason, brew felt it necessary to build all the dependencies for AARCH ARM64. I could do without that, I think...

**Reply by dtpoirot** [reply]
this is where I ended up  (in royal blue)
![IMG_2454](https://github.com/ImageMagick/ImageMagick/assets/3738983/802f5ac8-5002-420e-8b23-990807afb638)

**Reply by fmw42** [reply]
Post the original and specify your command(s).  Perhaps we can suggest other approaches.

**Reply by dtpoirot** [reply]
It's for a t-shirt design. This is blue ink silk-screened onto a light grey test sheet. There will be a second screen with text, printed in black. 

![image](https://github.com/ImageMagick/ImageMagick/assets/3738983/47ed5459-459b-4233-b04c-998ed53816d9)

**Comment by snibgo** [comment, marked as answer]
> How do I punch it up for screen printing (for a t-shirt)?

Define "punch it up". You might want to increase contrast of certain tonal ranges globally (using a power curve, or S-curve, or double-S-curve or whatever) or locally (see, for example, [Maximise local contrast](http://im.snibgo.com/maxloccont.htm)).

> I want to convert the RGB file to greyscale.

`-colorspace Gray`

> I want to normalize the histogram.

`-normalize`

> I want to clip and stretch the histogram.

`-level` or `+level`

> I may want to flatten the histogram.

Perhaps https://imagemagick.org/script/command-line-options.php#clahe . See also my [Contrast-limited equalisation](http://im.snibgo.com/eqlimit.htm), [Adaptive auto level and gamma](http://im.snibgo.com/algtile.htm).

> I then want to convert it to a pixelated halftone.

I show some halftone and [Lithographic effects](http://im.snibgo.com/lith.htm).
